airflow/utils/callback_requests.py

```diff
@@ -0,0 +1,101 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import datetime
+from typing import Optional
+
+from airflow.models.taskinstance import SimpleTaskInstance
+
+
+class CallbackRequest:
+    """
+    Base Class with information about the callback to be executed.
+
+    :param full_filepath: File Path to use to run the callback
+    :param msg: Additional Message that can be used for logging
+    """
+
+    def __init__(self, full_filepath: str, msg: Optional[str] = None):
+        self.full_filepath = full_filepath
+        self.msg = msg
+
+    def __eq__(self, other):
+        return self.__dict__ == other.__dict__
+
+    def __repr__(self):
+        return str(self.__dict__)
+
+
+class TaskCallbackRequest(CallbackRequest):
+    """
+    A Class with information about the success/failure TI callback to be executed. Currently, only failure
+    callbacks (when tasks are externally killed) and Zombies are run via DagFileProcessorProcess.
+
+    :param full_filepath: File Path to use to run the callback
+    :param simple_task_instance: Simplified Task Instance representation
+    :param is_failure_callback: Flag to determine whether it is a Failure Callback or Success Callback
+    :param msg: Additional Message that can be used for logging to determine failure/zombie
+    """
+
+    def __init__(
+        self,
+        full_filepath: str,
+        simple_task_instance: SimpleTaskInstance,
+        is_failure_callback: Optional[bool] = True,
+        msg: Optional[str] = None
+    ):
+        super().__init__(full_filepath=full_filepath, msg=msg)
+        self.simple_task_instance = simple_task_instance
+        self.is_failure_callback = is_failure_callback
+
+
+class DagCallbackRequest(CallbackRequest):
+    """
+    A Class with information about the is_failure_callback/failure DAG callback to be executed.
```

Comment:
I think one of those "failures" is meant to be success.

Reply:
fixed in https://github.com/astronomer/airflow/pull/913/commits/29af60d324509c46f983559b5ab6d702a42eebee